File: MediTrack.API/DTOs/Prescription/PrescriptionDto.cs
using System.ComponentModel.DataAnnotations;
using MediTrack.API.Models;

namespace MediTrack.API.DTOs.Prescription;

public class PrescriptionDto
{
    [Required]
    public string Dosage { get; set; } = string.Empty;
    
    public Frequency Frequency { get; set; }
    [Required]
    public DateTime StartDate { get; set; }
    public int Period { get; set; }
    public bool IsFlexible { get; set; } = false;
    public string Comment { get; set; } = string.Empty;
    [Range(1, int.MaxValue, ErrorMessage = "PatientId must be valid")]
    public int PatientId { get; set; }
    [Range(1, int.MaxValue, ErrorMessage = "MedicationId must be valid")]
    public int MedicationId { get; set; }
}

File: frontend/src/entities/prescriptions/prescription.dto.ts
export type Frequency = 
  | 'OnceADay'
  | 'TwiceADay' 
  | 'ThriceADay'
  | 'EveryOtherDay'
  | 'Weekly'

export interface PrescriptionRequestDto {
  dosage: string;
  frequency: Frequency; 
  startDate: string; 
  period: number;
  isFlexible: boolean;
  comment: string;
  patientId: number;
  medicationId: number;
}

export interface PrescriptionResponseDto {
  id: number;
  dosage: string;
  frequency: Frequency; 
  startDate: string; 
  period: number;
  isFlexible: boolean;
  comment: string;
  patientName: string;
  medicationName: string;
  doctorName: string;
}

export interface UpdatePrescriptionDto {
  dosage: string
  frequency: Frequency
  startDate: string
  period: number
  isFlexible: boolean
  comment: string
}


